File: nextGreatestLetter.py

# leetcode.com 744. Find Smallest Letter Greater Than Target

class Solution:
    def nextGreatestLetter(self, letters, target):
        """
        :type letters: List[str]
        :type target: str
        :rtype: str
        """
        pos = bisect.bisect_right(letters, target)
        return letters[0] if pos == len(letters) else letters[pos]

File: nextGreatestLetter.cpp

// leetcode.com  744. Find Smallest Letter Greater Than Target

#include <iostream>
#include <algorithm>
#include <vector>

using namespace std ;

class Solution {
public:
    char nextGreatestLetter(vector<char>& letters, char target) {
        auto it = std::upper_bound(letters.begin(), letters.end(), target) ;  

        if (it == letters.end()) it = letters.begin() ;

        return *it ;  
    }
};

